drivers/nvdimm/pmem.c

```diff
@@ -120,60 +483,38 @@ static blk_status_t pmem_clear_poison(struct pmem_device *pmem,
 static void write_pmem(void *pmem_addr, struct page *page,
 		unsigned int off, unsigned int len)
 {
-	unsigned int chunk;
 	void *mem;
 
-	while (len) {
-		mem = kmap_atomic(page);
-		chunk = min_t(unsigned int, len, PAGE_SIZE - off);
-		memcpy_flushcache(pmem_addr, mem + off, chunk);
-		kunmap_atomic(mem);
-		len -= chunk;
-		off = 0;
-		page++;
-		pmem_addr += chunk;
-	}
+	BUG_ON(off + len > PAGE_SIZE);
+	mem = kmap_atomic(page);
```

Comment:
Just for info: https://lwn.net/Articles/836144/.

Reply:
I'll have read the article, not sure what are you saying yet.  But this code was here before me.

Reply:
Was just saying ```kmap_local_page()``` is a newer way to map the user pages.

Reply:
Thanks Ameer. Good point. Updated.